frontend/controller/actions/group.js

```diff
@@ -131,6 +181,101 @@ export default (sbp('sbp/selectors/register', {
   'gi.actions/group/switch': function (groupId) {
     sbp('state/vuex/commit', 'setCurrentGroupId', groupId)
   },
+  'gi.actions/group/addChatRoom': async function (params: GIActionParams) {
+    const message = await sbp('gi.actions/chatroom/create', { data: params.data })
+
+    await sbp('chelonia/out/actionEncrypted', {
+      ...params,
+      action: 'gi.contracts/group/addChatRoom',
+      data: {
+        ...params.data,
+        chatRoomID: message.contractID()
+      }
+    })
+
+    return message
+  },
+  'gi.actions/group/joinChatRoom': async function (params: GIActionParams) {
+    try {
+      const rootState = sbp('state/vuex/state')
+      const username = params.data.username || rootState.loggedIn.username
+      await sbp('gi.actions/chatroom/join', {
+        contractID: params.data.chatRoomID,
+        data: { username }
+      })
+
+      if (username === rootState.loggedIn.username) {
+        sbp('okTurtles.data/set', 'READY_TO_JOIN_CHATROOM', true)
+      }
+      await sbp('chelonia/out/actionEncrypted', {
+        ...params,
+        action: 'gi.contracts/group/joinChatRoom'
+      })
+      sbp('okTurtles.data/set', 'READY_TO_JOIN_CHATROOM', false)
+    } catch (e) {
+      console.error('gi.actions/group/joinChatRoom failed!', e)
+      throw new GIErrorUIRuntimeError(L('Failed to join chat channel.'))
+    }
+    return true
+  },
+  'gi.actions/group/addAndJoinChatRoom': async function (params: GIActionParams) {
+    const message = await sbp('gi.actions/group/addChatRoom', {
+      ...params, hooks: { prepublish: params.hooks?.prepublish, postpublish: null }
+    })
+
+    await sbp('gi.actions/group/joinChatRoom', {
+      ...params,
+      data: { chatRoomID: message.contractID() },
+      hooks: { prepublish: null, postpublish: params.hooks?.postpublish }
+    })
+
+    return message
+  },
+  'gi.actions/group/renameChatRoom': async function (params: GIActionParams) {
+    try {
+      await sbp('gi.actions/chatroom/rename', {
+        contractID: params.data.chatRoomID,
+        data: { name: params.data.name }
```

Comment:
missing hook and publishOptions here too. Should use `..params` + override like you did it in `addAndJoinChatRoom`

Reply:
Done.